app/models/renewal/CashPayments.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.renewal
+
+import play.api.libs.json.{Json, Reads, Writes, __}
+
+case class CashPayments(cashPaymentsCustomerNotMet: CashPaymentsCustomerNotMet, howCashPaymentsReceived: Option[HowCashPaymentsReceived])
+
+object CashPayments {
+
+  implicit val jsonReads: Reads[CashPayments] = {
+    import play.api.libs.functional.syntax._
+    ((__ \ "receivePayments").read[Boolean] map CashPaymentsCustomerNotMet.apply and
+      (__ \ "paymentMethods").readNullable[PaymentMethods].map {
+        case Some(methods) => Some(HowCashPaymentsReceived.apply(methods))
+        case None => None
```

Comment:
Just a question but is `case None =>...` correct or would `case _ => ...` be more belt and braces?

Reply:
sure I'll change that